specs/openapi-spec.yaml

```diff
@@ -0,0 +1,322 @@
+openapi: 3.0.2
+info:
+  title: ION Storage
+  version: '${project.version}'
+  description: ION storage abstraction service
+  contact:
+    name: 'Connexta, LLC'
+    url: 'https://connexta.com'
+  license:
+    name: GNU Lesser General Public License
+    url: 'http://www.gnu.org/licenses/lgpl.html'
+servers:
+- url: '${rest.server.url}'
+
+paths:
```

Comment:
❓ Do we need to support a move file operation?
CDR will have the idea of a quarantined storage, which I assume in this API we will just mark as a separate directory, then if a file passes all checks it would move it to a separate storage that allows retrieval.
@emmberk

Reply:
We don't need these capabilities:
- list folders
- list folder's contents
- renames folder

It would also be nice to have the follow capabilities:
- upload file does not require a filename and, instead, the generated unique id is returned in the response. Right now, we will need to generate a unique identifier anyway instead of using the real filename.
- upload file returns a retrieve URL (so that CDR Team and Transform Team don't need to implement our own retrieve endpoint--there might be some authorization concerns with this)
- move file from one folder to another. I would expect this to be more performant re-uploading the file).

Lastly, this spec should include constraints on filename (e.g. 1-256 characters, regex) and file size (up to 10 GB).